### ISSUE TITLE
Issue with V1.22

### ISSUE THREAD

**Issue by JenNelsonChicago** [afawcett/declarative-lookup-rollup-summaries#235]
Hi, Andy - 

I am trying to deploy a child trigger from Campaign Member to Campaign and am getting this error on deploy:
dlrs_CampaignMemberTest.testTrigger System.DmlException: Insert failed. First exception on row 0; first error: MALFORMED_ID, Invalid campaign id: null: [CampaignId] (dlrs) Class.dlrs_CampaignMemberTest.testTrigger: line 11, column 1

**Comment by JenNelsonChicago** [comment]
Apologies, I know that isn't much to go on...but it is all I've got. :-)  I even tried using the new checkbox for Test Code See All Data and that didn't work either...same error.

**Comment by wes1278** [comment]
Hey @JenNelsonChicago ,

Can you please describe what you're trying to roll up? My gut reaction to your error is that you're field values in your rollup are telling the tool to do something it doesn't understand.

Can you take a picture of your rollup?

**Comment by JenNelsonChicago** [comment]
Hi, @wes1278 -

Thanks for looking.

I am trying to rollup a Count of Campaign Members with particular Statuses into a numeric field on Campaign.

The query that returns the correct count in Workbench SOQL Query is 
SELECT CampaignId,Id,Status FROM CampaignMember WHERE Status != 'Sent' AND Status != 'Responded' AND Status != 'Donated' AND Status != 'Removed'

Here's the screen shot of my rollup:

![image](https://cloud.githubusercontent.com/assets/10759854/9506015/2d3815c4-4c0b-11e5-9eb0-46144dfac568.png)

And here's the error I get when I try to deploy the trigger:

![image](https://cloud.githubusercontent.com/assets/10759854/9506054/591bb0f6-4c0b-11e5-9e60-a2828829e2de.png)

**Comment by afawcett** [comment]
This looks like its related to this https://github.com/afawcett/declarative-lookup-rollup-summaries/issues/197. Also do you have any Apex Triggers on CampaignMember?

**Comment by JenNelsonChicago** [comment]
Hi, @afawcett - we do have a trigger on CampaignMember for this package (http://sforce.co/1hKaKQG) from the AppExchange.  It allows us to get past the ever-present default Campaign Members Statuses "Sent" and "Responded".

**Comment by afawcett** [comment]
Yeah so it perhaps looks like you need the custom Apex test code snippet referenced in #197, though it seems the poster didn't share the code. So you need the new Test Code feature route and possibly if you want to take a less best practice route the see all data flag.

I've reluctantly added the new section on the rollup summary (you'll have to add the fields yourself if your upgrading) to allow for some customisation of the generated Apex Test code. In some cases, like yours, the rather dumb, insert the child record without any fields to trigger the rollup code for code coverage, doesn't work....

Thus we have to dip into some code land (hence my reluctancy). So in your case, you could probably tick the SeeAllData checkbox (this makes the test dependent on org data, not ideal if things change in the org the test breaks). Then you would paste something like this into the Test Code field and try Manage Child Trigger again (note you should see your overriden test code in the preview before hitting Deploy). 

    dlrs.RollupService.testHandler(new CampaignMember(CampaignId = 'someidthatexists'));

If you wanted a more robust route, don't tick the See All Data field and try something like this...

    Campaign newCampaign = new Campaign(); // Not sure what required fields it has?
    insert newCampaign;
    dlrs.RollupService.testHandler(new CampaignMember(CampaignId = newCampaign.Id));

I'm sorry we've had to get into code here, i've still not lost hope i could take some of these needs and make a declarative solution and perhaps capture more use cases like this, perhaps not all, but some more.

Let me know how you get on with this.

**Comment by afawcett** [comment]
Good news the poster on #197 has just responded to say they will share the code. So if the above doesn't work for you, hopefully their snippet will (assuming your scenarios and installed software are the same). Lets see.... :+1:

**Comment by afawcett** [comment]
@JenNelsonChicago hope you got this going, let me know if not. :+1:

**Comment by caitlinzinda** [comment]
Hola! I pasted in this code and it worked like a charm. I think screen shots would be super helpful as I was a little confused on where to paste the code and what certain text meant. So here it goes. I pasted the additional snippet 

Campaign c=new Campaign(Name='Test');
insert c;
Contact con=new Contact(Lastname='fred');
insert con;

// Force the dlrs_CampaignMemberTrigger to be invoked, fails the test if org config or other Apex code prevents this.
SavePoint sp = Database.setSavepoint();
dlrs.RollupService.testHandler(
    new CampaignMember(campaignid=c.id, Contactid=con.id));
Database.rollback(sp);

In the 
Child Object text box 

![Test Code Snippet Screen shot Campaign Members ](https://github.com/SFDO-Community/declarative-lookup-rollup-summaries/assets/87989138/435dbf64-0e6d-4658-9561-9043daa9aecd)

This solved the issue! Thanks all!
